pkg/process/v5/transformers/nvd/unique_pkg.go

```diff
@@ -63,14 +63,74 @@ func findUniquePkgs(cfgs ...nvd.Configuration) uniquePkgTracker {
 	return set
 }
 
-func determinePlatformCPEAndNodes(c nvd.Configuration) (*string, []nvd.Node) {
-	var platformCPE *string
+func platformPackageCandidates(set uniquePkgTracker, c nvd.Configuration) bool {
+	nodes := c.Nodes
+	/*
+		Turn a configuration like this:
+		(AND (redis <= 6.1 (OR debian:8 debian:9 ubuntu:19 ubuntu:20))
+		Into a configuration like this:
+		(OR (AND redis <= 6.1 debian:8) (AND redis <= 6.1 debian:9) (AND redis <= 6.1 ubuntu:19) (AND redis <= 6.1 ubuntu:20))
+	*/
+	if len(nodes) != 2 || c.Operator == nil || *c.Operator != nvd.And {
+		return false
+	}
+	var platformsNode nvd.Node
+	var applicationNode nvd.Node
+	for _, n := range nodes {
+		for _, c := range n.CpeMatch {
+			if isApplicationCPE(c) {
+				applicationNode = n
+				break
+			}
+			if isPlatformCPE(c) {
+				platformsNode = n
+				break
+			}
+		}
+	}
+	if platformsNode.Operator != nvd.Or || len(platformsNode.CpeMatch) < 2 {
+		return false
+	}
+	if applicationNode.Operator != nvd.Or {
+		return false
+	}
```

Comment:
this section could be broken into its own function that returns a boolean indicating if it is the expression pattern you're looking for... something like `func isPairwiseCPEConstraint(c nvd.Configuration) bool`

Reply:
Thanks for the suggestion! I hadn't done that because this block of code _also_ detects the order of the platform node vs the application node, and I couldn't think of a control flow that looked nicer than this.

@wagoodman Do you think it's important to keep looking for a way to refactor this?

Reply:
no, I think this is a great start, thanks for the effort 👍